packages/salesforcedx-vscode-soql/test/vscode-integration/testUtilities.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { AuthInfo, ConfigAggregator, Connection } from '@salesforce/core';
+import { JsonMap } from '@salesforce/ts-types';
+import { DescribeSObjectResult, QueryResult } from 'jsforce';
+import { SinonSandbox } from 'sinon';
+import * as vscode from 'vscode';
+import {
+  SoqlEditorEvent,
+  SOQLEditorInstance
+} from '../../src/editor/soqlEditorInstance';
+
+export interface MockConnection {
+  authInfo: object;
+  describeGlobal: () => Promise<void>;
+  describe: () => Promise<void>;
+  query: () => Promise<QueryResult<JsonMap>>;
+}
+
+export const mockQueryText = 'SELECT A, B FROM C';
+export const mockQueryData: QueryResult<JsonMap> = {
+  done: true,
+  totalSize: 2000,
+  records: [
+    {
+      attributes: {
+        type: 'C',
+        url: ''
+      },
+      A: '',
+      B: false
+    },
+    {
+      attributes: {
+        type: 'C',
+        url: ''
+      },
+      A: '',
+      B: false
+    },
+    {
+      attributes: {
+        type: 'C',
+        url: ''
+      },
+      A: '',
+      B: false
+    }
+  ]
+};
+
+export function getMockConnection(
+  sandbox: SinonSandbox,
+  testUserName = 'test@test.com'
+) {
+  const mockAuthInfo = { test: 'test' };
+  const mockConnection = {
```

Comment:
Not necessary now but you could explicitly type this object as a Connection and ts-ignore it in just this function so that you don't have to do it in individual test cases when you use it.